### PR TITLE
Add support for in-repo .gitflow config file

### PR DESCRIPTION
**Pull request by sometimesfood**
Store gitflow config in a file called `.gitflow`. That way `git flow init` settings can be version controlled and published.

This is a simple fix for nvie/gitflow#95.

9c36d66e137d7c22edd18115b2a57dc66f49eabd introduces a function called `gitflow_config_get`, whereas 9481814eae9da84caf3ea8e41389d41ace1ae607 introduces `gitflow_config_set`. The other commits simply modify gitflow to actually use these functions instead of calling git directly.

If you see any problems with this approach, just let me know.

### CODE REVIEW

**Comment by gvangool** [comment]
So you always write to `.gitflow`, but for backwards compatibility you still read from `.git/config`?

**Comment by sometimesfood** [comment]
Actually the main reason I'm still calling `git config` as a fallback is because I like to use my global git config for repos that don't have a `.gitflow`.

(`git config --get` also takes `~/.gitconfig` (used by `git config --global`) into account.)

The resulting backwards compatibility is nice to have, though.

File: gitflow-common [hunk 1]
```diff
@@ -152,39 +152,66 @@ git_is_branch_merged_into() {
...
+	[ $success -eq 0 ] && echo $value
+	return $success
+}
+
+# save config values to .gitflow
+gitflow_config_set() {
+	local key=$1
+	local value=$2
+	local config=$(gitflow_config_path)
+	git config --file "$config" $key "$value"
```
Comment (bobthecow) [inline gitflow-common]: I'd be more comfortable with something like this here:

```
if [ -f "$config" ]; then
    git config --file "$config" $key "$value" 
else
    git config $key "$value"
fi
```

Then it doesn't force the use of `.gitflow`, just like you wouldn't force the use of `.gitignore` or `.gitmodules` in a repo.

Comment (bobthecow) [inline gitflow-common]: hmm...

i guess that makes the init process a little weird though:

```
touch .gitflow
git flow init
```

... maybe add an option to `git flow init`?

Comment (sometimesfood) [inline gitflow-common]: Good point, I had thought about that too. In the end I opted for the current version in order to keep the commit simple. I figured @nvie could always just cherry-pick the completely non-invasive `gitflow_config_get` commits and we could find a better solution for the writes later on.

While I'm not especially fond of your first suggestion, I rather like the idea of a `git flow init` option. That would be much better than my current version. How about `-p` for _publish_?

**Comment by putermancer** [comment]
@sometimesfood Your latest commit causes errors when using git flow init without the publish flag. Specifically, the syntax to `git config` was wrong. I've forked your branch and added a commit to resolve this issue.

Feel free to merge the change from my branch into yours and add it to this pull request. With that change, this branch seems to be working as expected whether publishing the config or not.

The commit is 7a92aac627eb20009cbd01fdfc605d2206902402

**Comment by sometimesfood** [comment]
@bloveridge Oops, must have pushed an incomplete branch to GitHub. Thanks for the fix.

**Comment by petervanderdoes** [comment]
@sometimesfood @bloveridge 
Based on the comment https://github.com/nvie/gitflow/pull/184#issuecomment-3326352 I think it won't be implemented in the nvie/gitflow development.
Issue #184 was based upon this pull request but it used a separate directory for the configuration and future implementation of hooks.

**Comment by sometimesfood** [comment]
@petervanderdoes: Apparently you're right. Closing this.

**Comment by baquiax** [comment]
Seems like #184 has an issue. :( 

I can't read the context. 

image

**Pull request closed without merging**